docs/source/queries/paged.md

```diff
@@ -197,5 +217,18 @@ loop {
 ```
 
 ### Performance
-Performance is the same as in non-paged variants.\
-For the best performance use [prepared queries](prepared.md).
\ No newline at end of file
+For the best performance use [prepared queries](prepared.md).
+See [query types overview](queries.md).
+
+## Best practices
+
+| Query result fetching   | Unpaged                                                                                                                 | Paged manually                                                                                       | Paged automatically                                                                               |
+|-------------------------|-------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------|
+| Exposed Session API     | `{query,execute}_unpaged`                                                                                               | `{query,execute}_single_page`                                                                        | `{query,execute}_iter`                                                                            |
+| Working                 | get all results in a single CQL frame, into a single Rust struct                                                        | get one page of results in a single CQL frame, into a single Rust struct                             | upon high-level iteration, fetch consecutive CQL frames and transparently iterate over their rows |
+| Cluster load            | potentially **HIGH** for large results, beware!                                                                         | normal                                                                                               | normal                                                                                            |
+| Driver overhead         | low - simple frame fetch                                                                                                | low - simple frame fetch                                                                             | considerable - `RowIteratorWorker` is a separate tokio task                                       |
+| Feature limitations     | none                                                                                                                    | none                                                                                                 | speculative execution not supported                                                               |
```

Comment:
I'm wondering if we should print warning / erorr query if speculative execution is set on execution profile used in {query ,execute}_iter.

Reply:
Hmm, that might be a good idea, but OTOH forcing using different profiles for iter and non-iter execution and thus being inconvenient (for speculative execution users).

Reply:
I worry about situation where user wants to use speculative execution,  but silently doesn't because of `_iter` methods. You raised good point, I'm not sure how to best handle this possible issue.